### PR TITLE
Custom evt

### PR DESCRIPTION
**Pull request by beliaev-maksim**
## Issue

user did have a possibility to change service definition during runtime. 

Now we allow user to add custom events to be listened

### CODE REVIEW

**Comment by beliaev-maksim** [comment]
@dstathis 
is it fine that I bumped the version or should it be done right before publishing the lib?

**Comment by dstathis** [comment]
> @dstathis 
> is it fine that I bumped the version or should it be done right before publishing the lib?

Yes the libpatch should be bumped. That way you can release right after merging.

**Review by dstathis — DISMISSED** [review]

**Comment by rbarry82** [comment]
LGTM in general (after the `LIBPATCH` bump). What's the use case?

**Comment by beliaev-maksim** [comment]
@rbarry82 
I want to have an app and expose port as a part of the configuration file. It works brilliantly with prometheus scrape job targets but service patch is not applied. Thus, app is not available on service IP:PORT

and there was no way to reapply patch to a new configuration

I would assume there might be other use cases when user will need to fire a patching on a custom event

**Review by sed-i — COMMENTED** [review]
LG2M.

However, this lib is unique: the change can be applied on [install](https://juju.is/docs/sdk/install-event), which is the fist event to be emitted in the lifecycle.

This means that charm code can do all its logic in the constructor, without subscribing to any events. In particular, charm code can use lightkube to check if the port arg matches the previously applied changes and act accordingly. Seems like the only _potential_ exception is the `remove` event.

**Review by sed-i — APPROVED** [review]

File: lib/charms/observability_libs/v1/kubernetes_service_patch.py
Comment (beliaev-maksim) [reply lib/charms/observability_libs/v1/kubernetes_service_patch.py:184]: But if user changes config, then install is not fired. I don't see a way to reapply the patch outside of this lib

Comment (sed-i) [reply lib/charms/observability_libs/v1/kubernetes_service_patch.py:184]: I'm saying that you probably don't even need `install`: seems like all logic can be run at once in the constructor.
But this would be a deviation from current practice and is outside the scope of this PR :)

Comment (beliaev-maksim) [reply lib/charms/observability_libs/v1/kubernetes_service_patch.py]: ahh, I see

**Review by sed-i — APPROVED** [review]

**Pull request merged**
